dandi/tests/test_organize.py

```diff
@@ -115,6 +116,31 @@ def test_organize_nwb_test_data(nwb_test_data, tmpdir, clirunner, mode):
         nwb_test_data = op.relpath(nwb_test_data, cwd)
         outdir = op.relpath(outdir, cwd)
 
+    src = Path(tmpdir, "src")
+    src.touch()
+    dest = Path(tmpdir, "dest")
+    try:
+        dest.symlink_to(src)
+    except OSError:
+        symlinks_work = False
+    else:
+        symlinks_work = True
+    try:
+        dest.unlink()
+    except FileNotFoundError:
+        pass
+    try:
+        os.link(src, dest)
+    except OSError:
+        hardlinks_work = False
+    else:
+        hardlinks_work = True
```

Comment:
I am ok to merge as is to progress.  

But I feel that we already have logic in multiple places, so it would be worthwhile to introduce some helpers. E.g. (design is just suggestive):
- a helper `get_filesystem_capabilities(dirpath, capabilities=('symlinks', 'hardlinks'))` which would may be return a set of supported capabilities.
- And some shim for `tmpdir` fixture (`tmpdir_caps`) fixture, which could 
  - enhance `tmpdir` with args `skip_missing=None` where we would list capabilities which if missing would cause test to be skipped
  - `tmpdir` provided structure should be enhanced with `.caps` or `.capabilities` with the set from above helper or add a set of `.supports_X` for X being each capability.
  then in each test we could avoid doing all such checks, but just `if tmpdir.supports_symlinks`

What do you think @jwodder ?

Reply:
Augmenting `tmpdir` seems a little like overkill at the moment.  Perhaps just using `supports_{sym,hard}links(dir)` helpers that wrap `get_filesystem_capabilities()` would be enough.  Other than that, sounds like a plan.

Reply:
ok, let's then proceed with merge of this PR for now as is. Helper could followup in its own PR